pulser-core/pulser/sequence/sequence.py

```diff
@@ -669,6 +804,13 @@ def declare_channel(
         if ch.basis == "XY" and not self._in_xy:
             self._in_xy = True
             self.set_magnetic_field()
+            # If schedule contains a DMM channel, delete it.
+            if len(self._schedule) > 0:
+                print(self._schedule)
+                assert len(self._schedule) == 1 and isinstance(
+                    list(self._schedule.values())[0], _DMMSchedule
+                )
+                self._schedule.pop(list(self._schedule.keys())[0])
```

Comment:
So you are allowing for a DMM to be declared and then deleting it if you go into XY mode? To me, declaring a DMM should tie you down to Ising mode from that moment on